dockerfiles/cli/scripts/cli.sh

```diff
@@ -128,39 +169,36 @@ cmd_start_check_ports() {
-  fi
-
-  if ! $(port_open 80) || \
-     ! $(port_open 443) || \
-     ! $(port_open 2181) || \
-     ! $(port_open 5000) || \
-     ! $(port_open 23750) || \
-     ! $(port_open 23751) || \
-     ! $(port_open 32001) || \
-     ! $(port_open 32101); then
-     echo ""
-     error "Ports required to run $CHE_MINI_PRODUCT_NAME are used by another program."
-     return 2;
-  fi
-  if debug_server; then
-    if ! $(port_open ${CODENVY_DEBUG_PORT}) || ! $(port_open 9000); then
-      echo ""
-      error "Ports required to run $CHE_MINI_PRODUCT_NAME are used by another program."
-      return 1;
+  if check_all_ports; then
+    text   "         port 80 (http):        ${GREEN}[AVAILABLE]${NC}\n"
+    text   "         port 443 (https):      ${GREEN}[AVAILABLE]${NC}\n"
+    text   "         port 2181 (zookeeper): ${GREEN}[AVAILABLE]${NC}\n"
+    text   "         port 5000 (registry):  ${GREEN}[AVAILABLE]${NC}\n"
+    text   "         port 23750 (socat):    ${GREEN}[AVAILABLE]${NC}\n"
+    text   "         port 23751 (swarm):    ${GREEN}[AVAILABLE]${NC}\n"
+    text   "         port 32001 (jmx):      ${GREEN}[AVAILABLE]${NC}\n"
+    text   "         port 32101 (jmx):      ${GREEN}[AVAILABLE]${NC}\n"
+    if debug_server; then
+      text   "         port $CODENVY_DEBUG_PORT (debug):     ${GREEN}[AVAILABLE]${NC}\n"
+      text   "         port 9000 (lighttpd):  ${GREEN}[AVAILABLE]${NC}\n"
+    fi
+  else
+    text   "         port 80 (http):        $(port_open 80 && echo "${GREEN}[AVAILABLE]${NC}" || echo "${RED}[ALREADY IN USE]${NC}") \n"
+    text   "         port 443 (https):      $(port_open 443 && echo "${GREEN}[AVAILABLE]${NC}" || echo "${RED}[ALREADY IN USE]${NC}") \n"
+    text   "         port 2181 (zookeeper): $(port_open 2181 && echo "${GREEN}[AVAILABLE]${NC}" || echo "${RED}[ALREADY IN USE]${NC}") \n"
+    text   "         port 5000 (registry):  $(port_open 5000 && echo "${GREEN}[AVAILABLE]${NC}" || echo "${RED}[ALREADY IN USE]${NC}") \n"
+    text   "         port 23750 (socat):    $(port_open 23750 && echo "${GREEN}[AVAILABLE]${NC}" || echo "${RED}[ALREADY IN USE]${NC}") \n"
+    text   "         port 23751 (swarm):    $(port_open 23751 && echo "${GREEN}[AVAILABLE]${NC}" || echo "${RED}[ALREADY IN USE]${NC}") \n"
+    text   "         port 32001 (jmx):      $(port_open 32001 && echo "${GREEN}[AVAILABLE]${NC}" || echo "${RED}[ALREADY IN USE]${NC}") \n"
+    text   "         port 32101 (jmx):      $(port_open 32101 && echo "${GREEN}[AVAILABLE]${NC}" || echo "${RED}[ALREADY IN USE]${NC}") \n"
+    if debug_server; then
+      text   "         port ${CODENVY_DEBUG_PORT} (debug):     $(port_open ${CODENVY_DEBUG_PORT} && echo "${GREEN}[AVAILABLE]${NC}" || echo "${RED}[ALREADY IN USE]${NC}") \n"
+      text   "         port 9000 (lighttpd):  $(port_open 9000 && echo "${GREEN}[AVAILABLE]${NC}" || echo "${RED}[ALREADY IN USE]${NC}") \n"
```

Comment:
I added a first pass of the improvement.

I am not quite sure how to handle it.  But I think we can probably put this array into `cmd_pre_init` for a custom CLI assembly and then this would probably remove the need for the custom port check methods in all of the various CLIs.